doc/over_react_redux_documentation.md

```diff
@@ -0,0 +1,320 @@
+    also expensive to run, passing in a custom function for `areStatesEqual` may be beneficial to performance.
+- #### `areOwnPropsEqual` 
+    - Does a shallow Map equality check by default.
+    - Takes a function with the signature `(next: TProps, prev: TProps) => bool`.
+    - When [pure](#pure), compares incoming props to its previous value. This comparison can be used to conduct 
+    operations such as being part of a process to whitelist certain props.
+- #### `areStatePropsEqual` 
+    - Does a shallow Map equality check by default.
+    - Takes a function with the signature `(next: TProps, prev: TProps) => bool`.
+    - When [pure](#pure), compares the results of `mapStateToProps` with its previous value. Similar to other 
+    equality check callbacks, this provides the opportunity to optimize performance.
+- #### `areMergedPropsEqual` 
+    - Does a shallow Map equality check by default.
+    - Takes a function with the signature `(next: TProps, prev: TProps) => bool`.
+    - Can be passed in for small performance improvements. Two possible cases are to implement `deepEqual` or 
+    `strictEqual`.
+        > See: <https://react-redux.js.org/api/connect#aremergedpropsequal-next-object-prev-object-boolean>
+- #### `context` 
+    - Can be utilized to provide a custom context object created with `createContext`.
+    - `context` can be used to utilize multiple stores. 
+    > For more information on having multiple stores, see [Using Multiple Stores](#using-multiple-stores).
+- #### `pure` 
+    - If `true` (default), connect performs several equality checks that are used to avoid unnecessary
+calls to `mapStateToProps`, `mapDispatchToProps`, `mergeProps`, and ultimately to `render`. These include
+`areStatesEqual`, `areOwnPropsEqual`, `areStatePropsEqual`, and `areMergedPropsEqual`.
+While the defaults are probably appropriate 99% of the time, you may wish to override them with custom
+implementations for performance or other reasons.
+- #### `forwardRef` 
+    - If `true`, the `ref` prop provided to the connected component will be return the wrapped component.
+
+> [More information about the `connect` function](https://react-redux.js.org/api/connect#connect)
+  
+## Using Multiple Stores
+An application can have multiple stores by both utilizing the `context` prop of `ReduxProvider` and setting the 
+`context` parameter on `connect`. While this is possible, it is not recommended.
+> See: <https://redux.js.org/api/store#a-note-for-flux-users>
+
+> See: <https://stackoverflow.com/questions/33619775/redux-multiple-stores-why-not>
+
+__Multiple Stores Example:__
+```dart
+Store store1 = new Store<CounterState>(counterStateReducer, initialState: new CounterState(count: 0));
+Store store2 = new Store<BigCounterState>(bigCounterStateReducer, initialState: new BigCounterState(bigCount: 100));
+
+UiFactory<CounterProps> ConnectedCounter = connect<CounterState, CounterProps>(
+      mapStateToProps: (state) => (Counter()..count = state.count)
```

Comment:
#nit would be nice to have this be a two-space indent with a trailing comma to match the one below
```suggestion
    mapStateToProps: (state) => (Counter()..count = state.count),
```